src/main/java/filter/Parser.java

```diff
@@ -173,13 +174,25 @@ private FilterExpression parseQualifier(Token token) {
         qualifierName = qualifierName.substring(0, qualifierName.length() - 1).trim();
 
         QualifierType type = QualifierType.parse(qualifierName).orElse(QualifierType.FALSE);
-        return parseQualifierContent(type, false);
+        if (type == QualifierType.SORT) {
+            //special case for sort
```

Comment:
Because the semicolon operator doesn't make sense only when applied to sort qualifier.

We can also just parse it as it is and leave it to `UpdateController` to determine what to do when there are more than 1 sort qualifier (#1173?). Either that or we can prevent it from the parser by raising `ParseException`.

I just realised that I forgot to set the priority of semicolon type to handle a few corner cases. Will fix them along with this issue and write some tests as well.